core/src/main/java/brooklyn/BrooklynVersion.java

```diff
@@ -19,80 +19,218 @@
+              // allowed for dev env
+              log.trace("No maven resource file "+MVN_VERSION_RESOURCE_FILE+" available");
+          } else {
+              log.warn("No maven resource file "+MVN_VERSION_RESOURCE_FILE+" available");
+          }
+          return;
+      }
       versionProperties.load(checkNotNull(versionStream));
-    } catch (IOException exception) {
-      throw new IllegalStateException(format("Unable to load version resource file '%s'", VERSION_RESOURCE_FILE), exception);
+    } catch (IOException e) {
+      log.warn("Error reading maven resource file "+MVN_VERSION_RESOURCE_FILE+": "+e, e);
     }
-    return checkNotNull(versionProperties.getProperty(VERSION_PROPERTY_NAME), VERSION_PROPERTY_NAME);
+  }
+
+  /** reads properties from brooklyn-core's manifest */
+  private void readPropertiesFromOsgiResource(ClassLoader resourceLoader, String symbolicName) {
+      Enumeration<URL> paths;
+      try {
+          paths = BrooklynVersion.class.getClassLoader().getResources(MANIFEST_PATH);
+      } catch (IOException e) {
+          // shouldn't happen
+          throw Exceptions.propagate(e);
+      }
+      while (paths.hasMoreElements()) {
+          URL u = paths.nextElement();
+          try {
+              ManifestHelper mh = Osgis.ManifestHelper.forManifest(u.openStream());
+              if (BROOKLYN_CORE_SYMBOLIC_NAME.equals(mh.getSymbolicName())) {
+                  Attributes attrs = mh.getManifest().getMainAttributes();
+                  for (Object key: attrs.keySet()) {
+                      // key is an Attribute.Name; toString converts to string
+                      versionProperties.put(key.toString(), attrs.getValue(key.toString()));
+                  }
+                  return;
+              }
+          } catch (Exception e) {
+              Exceptions.propagateIfFatal(e);
+              log.warn("Error reading OSGi manifest from "+u+" when determining version properties: "+e, e);
+          }
+      }
+      if (isDevelopmentEnvironment()) {
+          // allowed for dev env
+          log.trace("No OSGi manifest available to determine version properties");
+      } else {
+          log.warn("No OSGi manifest available to determine version properties");
+      }
+  }
+
+  /** 
+   * Returns whether this is a Brooklyn dev environment,
+   * specifically core/target/classes/ is on the classpath for the org.apache.brooklyn.core project.
+   * <p>
+   * In a packaged or library build of Brooklyn (normal usage) this should return false,
+   * and all OSGi components should be available.
+   */
+  public static boolean isDevelopmentEnvironment() {
+      Boolean isDevEnv = IS_DEV_ENV.get();
+      if (isDevEnv!=null) return isDevEnv;
+      synchronized (IS_DEV_ENV) {
+          isDevEnv = computeIsDevelopmentEnvironment();
+          IS_DEV_ENV.set(isDevEnv);
+          return isDevEnv;
+      }
+  }
+  
+  private static boolean computeIsDevelopmentEnvironment() {
```

Comment:
good spot, i'll clean this up.  (i hadn't meant to merge this yet... another fat-finger.)